Dockerfile

```diff
@@ -1,36 +1,43 @@
-FROM registry.access.redhat.com/ubi8/ubi-minimal as builder
-
-# create enviroment variables and files
-ENV ENVFILE=/tmp/envfile
-RUN echo "export ARCH=`uname -m | sed 's/x86_64/amd64/'`" >> ${ENVFILE}
-ENV PATH=$PATH:/usr/local/go/bin
+ARG TARGET_ARCH=amd64
+FROM registry.access.redhat.com/ubi8/ubi-minimal as base
+ARG TARGET_ARCH
 
+# download packages
 RUN microdnf install -y make tar gzip which && microdnf clean all
+RUN export ARCH=$(uname -m | sed 's/x86_64/amd64/'); curl -L https://go.dev/dl/go1.21.6.linux-${ARCH}.tar.gz | tar -C /usr/local -xzf -
```

Comment:
```suggestion
RUN ARCH=$(uname -m | sed 's/x86_64/amd64/') curl -L https://go.dev/dl/go1.21.6.linux-${ARCH}.tar.gz | tar -C /usr/local -xzf -
```

Reply:
This change does not work with my podman version. How did you test it?

Reply:
Ah yes, I see.

It would only work with something like this:

```
RUN ARCH=$(uname -m | sed 's/x86_64/amd64/') bash -c 'curl -L https://go.dev/dl/go1.21.6.linux-${ARCH}.tar.gz | tar -C /usr/local -xzf -'
```

Let's leave it like it is.